examples/split_plot_rcbd.R

```diff
@@ -0,0 +1,216 @@
+#(1) Create a sqrt-transformed variable
+my.data$Yield <- sqrt(my.data$Yield)
+
+#(2) Create a log-transformed variable
+my.data$Yield <- log10(my.data$Yield)
+
+#(3)----- Finding the exponent for a power transformation ---- #
+mean.data <- aggregate(Yield ~ SeedLotA + TrtmtB, data = my.data, function(x)
+                       c(logmean=log10(mean(x)), logvar=log10(var(x))))
+power.model <- lm(logvar ~ logmean, data = as.data.frame(mean.data$Yield))
+power <- 1 - summary(power.model)$coefficients[2, 1] / 2
+my.data$Yield <- my.data$Yield^power
+
+# This is the standard model for a split plot RCBD.
+# NOTE : I'm getting this warning message:
+# Warning message:
+# In aov(formula = Yield ~ SeedLotA + Block + Error(SeedLotA:Block) +  :
+#   Error() model is singular
+#-----------------------------------------------------------------------------#
+model <- aov(formula = Yield ~ SeedLotA + Block + Error(SeedLotA:Block) +
+             TrtmtB + SeedLotA:TrtmtB, data = my.data)
+#-----------------------------------------------------------------------------#
+
+# Following the advice in [2], I create the same model without the error term
+# for assumption testing (and post hoc?). The F values and P values are not
+# correct for this model so the ANOVA table should not be shown to the user.
+#-----------------------------------------------------------------------------#
+model.tmp <- aov(formula = Yield ~ Block + SeedLotA + TrtmtB + SeedLotA:Block +
+                 SeedLotA:TrtmtB, data = my.data)
+#-----------------------------------------------------------------------------#
+
+# Plot the four standard fit plots: residuals vs predicted and the Normal Q-Q
+# plot of the residuals.
+dev.new()
+#-----------------------------------------------------------------------------#
+par(mfrow = c(2, 1), oma = c(0, 0, 2, 0))
+plot(model.tmp, which = c(1, 2))
+#-----------------------------------------------------------------------------#
+invisible(dev.copy(png, 'split-plot-rcbd-fit-plots.png'))
+invisible(dev.off())
+
+# Do not do the Shapiro-Wilk Normality Test because it is not straight forward
+# for a split-plot design.
+
+# Levene's Test.
+cat("Levene's Test\n")
+sep(79)
+#-----------------------------------------------------------------------------#
+leveneTest(Yield ~ SeedLotA, data = my.data)
+leveneTest(Yield ~ TrtmtB, data = my.data)
+#-----------------------------------------------------------------------------#
+sep(79)
+
+# 1-df Tukey:
+cat("One DoF Tukey Test\n")
+sep(79)
+#-----------------------------------------------------------------------------#
+my.data$sq_preds <- predict(model.tmp)^2
+one.df.model <- lm(Yield ~ SeedLotA + Block + SeedLotA:Block + TrtmtB +
+                   SeedLotA:TrtmtB + sq_preds, my.data)
+anova(one.df.model)
+#-----------------------------------------------------------------------------#
+sep(79)
+
+# Print the ANOVA table of the fit. The user will have to note the significant
+# factors.
```

Comment:
I think we should make the post-hoc tests fixed based on the ANOVA output as outlined starting in line 145